feat(docs): Added behavior queue limit section to macros page (#1384)
zmkfirmware/zmk: docs/docs/config/behaviors.md

See the [zmk/app/dts/behaviors/](https://github.com/zmkfirmware/zmk/tree/main/app/dts/behaviors) folder for all default behaviors.

> ## Kconfig

Reviewer: It's tricky finding somewhere for `CONFIG_ZMK_BEHAVIORS_QUEUE_SIZE` to live because it doesn't actually apply to a single behavior specifically. I do think that here at the top of the Behaviors page is most appropriate place for it.

However, I think it might make sense to have it nested beneath a new heading like so:

```suggestion
## General

These are general settings that control behaviors at a higher level.

### Kconfig
```

This would bring it in alignment with other pages, which are typically structured like this (e.g. https://zmk.dev/docs/config/kscan):

> * Group 1
>   * Kconfig
>   * Devicetree
> * Group 2
>   * Kconfig
>   * Devicetree

This would make the hierarchy of content for this page like so:

> - General
> 	- Kconfig
> - Caps Word
> 	- Devicetree
> - Hold-Tap
> 	- Devicetree
> - Key Repeat
> 	- Devicetree
> - Macro
> 	- Devicetree
> - Mod-Morph
> 	- Devicetree
> - Sticky Key
> 	- Devicetree
> - Tap Dance
> 	- Devicetree

What do you think @GreenAirplane, @caksoylar, and @joelspadin?

Reviewer: Actually, to align it with the kscan page I linked to as an example ([source](https://github.com/zmkfirmware/zmk/blob/f05a7a13f2e73506eefb2fae06da4d95dbdd1ec0/docs/docs/config/kscan.md)), it could just simply be:

```suggestion
## Common

### Kconfig
```

Reviewer 2: Good suggestion! "Common" is good I'd say.

Author: Yeah, I think `Common` is the best option, especially considering that it is already used in https://zmk.dev/docs/config/kscan.

Committed the suggestion.

| Config                             | Type | Description                                          | Default |
| ---------------------------------- | ---- | ---------------------------------------------------- | ------- |

> | `CONFIG_ZMK_BEHAVIORS_QUEUE_SIZE`  | int  | The size of the behaviors queue                      | 64      |

Reviewer 2: ```suggestion
| `CONFIG_ZMK_BEHAVIORS_QUEUE_SIZE`  | int  | Maximum number of behaviors to allow queueing from a macro or other complex behavior | 64      |
```

The descriptions in these docs match the description in the Kconfig file. Note that you need to run `prettier` again to format the table properly after this change.

Author: done!
